www/unstable/userguide/weekyears.md

```diff
@@ -0,0 +1,291 @@
+  that year; otherwise, the whole of that week is in the final week
+  of the previous year.
+  
+To put the "four days or more" rule in a different way: if the first
+day of the calendar year is Monday, Tuesday, Wednesday or Thursday,
+then the week containing that day is week 1 of the same year; if
+the calendar year starts on a Friday, Saturday or Sunday, then that
+week is in the previous week-year.
+
+Examples:
+
+- January 1st 2013 is a Tuesday, so 2013 week 1 runs from 2012-12-31
+to 2013-01-06 inclusive
+- January 1st 2014 is a Wednesday, so 2014 week 1 runs from 2013-12-30
+to 2013-01-05 inclusive
+- January 1st 2015 is a Thursday, so 2015 week 1 runs from
+2014-12-29 to 2015-01-04 inclusive
+- January 1st 2016 is a Friday, so **2015 week 53** runs from
+2015-12-28 to 2016-01-03 inclusive, and 2016 week 1 runs from
+2016-01-04 to 2016-01-10 inclusive.
+- January 1st 2017 is a Sunday, so **2016 week 52** runs from
+2016-12-26 to 2017-01-01 inclusive, and 2017 week 1 runs from
+2017-01-02 to 2017-01-08 inclusive.
+
+Once you understand this, it's not terribly hard to add two other
+axes of flexibility:
+
+- First day of week: for example, if you want Sunday-to-Saturday
+weeks, but still "four days or more" then the first day of the
+calendar year is in week 1 if it's a Sunday, Monday, Tuesday or
+Wednesday.
+- The required number of days in the week containing the first day
+of the calendar year to make that day part of week 1. For example, 
+with a Monday-to-Sunday week, and a "minimum days in first week"
+value of 7, the first day of the calendar year will *only* be in
+week 1 if it's a Monday. At the other extreme, a "minimum days
+in first week" value of 1 means the first day of the calendar year
+is *always* in week 1.
+
+Use the `WeekYearRules.ForMinDaysInFirstWeek(int)` and
+`WeekYearRules.ForMinDaysInFirstWeek(int, IsoDayOfWeek)` methods to
+construct ISO-like rules. The first overload defaults to using Monday
+as the first day of the week.
+
+The rule returned by `WeekYearRules.Iso` property is equivalent to
+the rule returned by 
+`WeekYearRules.ForMinDaysInFirstWeek(4, IsoDayOfWeek.Monday)`.
+
+BCL rules
+---
+
+BCL rules are obtained via `WeekYearRules.FromBclRule`, which
+accepts two parameters:
+
+- A [`CalendarWeekRule`](https://msdn.microsoft.com/en-us/library/system.globalization.calendarweekrule)
+- A [`DayOfWeek`](https://msdn.microsoft.com/en-us/library/system.dayofweek)
+
+These mirror the second and third parameters to the
+[`Calendar.GetWeekOfYear`](https://msdn.microsoft.com/en-us/library/system.globalization.calendar.getweekofyear)
+method in the BCL.
+
+It's easiest to think of the BCL rules as being equivalent to the
+ISO-like rules in most respects, with a mapping from
+`CalendarWeekRule` to "minimum number of days in first week" of:
+
+- `FirstDay`: 1
+- `FirstFourDayWeek`: 4
+- `FirstFullWeek`: 7
+
+There's then one subtle difference between ISO-like rules and
+BCL-like rules:
```

Comment:
Perhaps add a summary here:

"
There's then one subtle difference between ISO-like rules and BCL-like rules: BCL weeks do not always have seven days.

> (quote)

"